### ISSUE TITLE
About how to use the final plan list after planning

### ISSUE THREAD

**Issue by Bijiazhou** [EricZinda/InductorHtn#2]
- At present, the final plan after HTN planning is a **huge sequence composed of all operators.** If it is adopted in the project, do I need to write a **specific interpreter** to connect the functions in the project? Or is there a more **efficient way**?

- In addition, I don't think HTN is particularly suitable for mathematical calculation, so should I put most of the contents related to numerical calculation in other places, mainly leaving the qualitative analysis part to simplify the process？

Could you please give me some advice？

**Comment by EricZinda** [comment]
Yes, the goal of the HTN Planner is to come up with a *plan* which is a set of operators.  You'll need to interpret these operators in the way that makes sense in your application.  [this post](https://blog.inductorsoftware.com/blog/htninexospecies) talks a little about how I did in for my strategy game Exospecies (look at step 3 in the post).  There is no way to simplify that step since it is all about your code and what those operations mean in your scenario.

Without knowing more about your scenario it is tough to answer your second question.  It is true that doing mathematical calculations is often best done outside of the engine if they are complex.  However, the basic stuff like addition, multiplication, etc can be done inside the engine.  I did use the engine for all the math I needed for my strategy game.  Your scenario may have more complex needs, though.
